storage/postgres/broker.go

```diff
@@ -17,37 +17,156 @@
 package postgres
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/Peripli/service-manager/types"
 	"github.com/jmoiron/sqlx"
-	"context"
+	"github.com/sirupsen/logrus"
 )
 
 type brokerStorage struct {
 	db *sqlx.DB
 }
 
-func (storage *brokerStorage) Create(ctx context.Context, broker *types.Broker) error {
-	return nil
+func (bs *brokerStorage) Create(broker *types.Broker) error {
+	return transaction(bs.db, func(tx *sqlx.Tx) error {
+		credentialsDTO := convertCredentialsToDTO(broker.Credentials)
+		statement, err := tx.PrepareNamed(
+			"INSERT INTO " + credentialsTable + " (type, username, password, token) VALUES (:type, :username, :password, :token) RETURNING id")
+		if err != nil {
+			logrus.Error("Unable to create prepared statement")
+			return err
+		}
+
+		var credentialsID int
+		err = statement.Get(&credentialsID, credentialsDTO)
+		if err != nil {
+			logrus.Error("Prepared statement execution failed")
+			return err
+		}
+		brokerDTO := convertBrokerToDTO(broker)
+		brokerDTO.CredentialsID = credentialsID
+
+		_, err = tx.NamedExec(fmt.Sprintf(
+			"INSERT INTO %s (id, name, description, broker_url, created_at, updated_at, credentials_id) %s",
+			brokerTable,
+			"VALUES (:id, :name, :description, :broker_url, :created_at, :updated_at, :credentials_id)"),
+			&brokerDTO)
+		return checkUniqueViolation(err)
+	})
+}
+
+func (bs *brokerStorage) Get(id string) (*types.Broker, error) {
+	broker := &struct {
+		*Broker
+		Username string `db:"username"`
+		Password string `db:"password"`
+	}{
+		&Broker{}, "", "",
+	}
+
+	query := fmt.Sprintf(`SELECT b.id, b.name, b.description, b.broker_url, b.created_at, b.updated_at, c.username, c.password
+						 FROM %s AS b INNER JOIN %s AS c ON (b.credentials_id=c.id)
+						 WHERE b.id=$1`, brokerTable, credentialsTable)
+
+	err := bs.db.Get(broker, query, id)
+
+	if err != nil {
+		return nil, checkSQLNoRows(err)
+	}
+
+	result := broker.Convert()
+	result.Credentials = types.NewBasicCredentials(broker.Username, broker.Password)
+
+	return result, nil
 }
 
-func (storage *brokerStorage) Find(ctx context.Context, id string) (*types.Broker, error) {
-	return &types.Broker{
-		Name:     "brokerName",
-		ID:       "brokerID",
-		URL:      "http://localhost:8080/broker",
-		User:     "brokerAdmin",
-		Password: "brokerAdmin",
-	}, nil
+func (bs *brokerStorage) GetAll() ([]types.Broker, error) {
+	brokers := []Broker{}
+	err := bs.db.Select(&brokers, "SELECT id, name, description, created_at, updated_at, broker_url FROM "+brokerTable)
+	if err != nil {
+		logrus.Error("An error occurred while retrieving all brokers")
+		return nil, err
+	}
+	restBrokers := make([]types.Broker, len(brokers))
```

Comment:
note for future improvement - if this remains as is, the len should actually be cap.